contracts/RootChain.sol

```diff
@@ -481,38 +483,41 @@ contract RootChain {
         InFlightExit storage inFlightExit = _getInFlightExit(_inFlightTx);
         require(inFlightExit.exitStartTimestamp == 0);
 
-        // Get information about the outputs.
-        uint8 numInputs;
-        uint256 outputSum;
-        // TODO: re-write to support ERC20 tokens
-        (numInputs, outputSum) = _getOutputInfo(_inFlightTx);
+        // Check if such an in-flight exit has already been finalized
+        require(!flagged(inFlightExit.exitMap));
```

Comment:
I am actually confused here. Might be some newbie dumb question but when would `inflightExit.exitMap` being `flagged` (as per my understanding, the left-most bit being set)?

Since there is no document on `exitMap` so I tried to trace the code myself, it seems like only the right-most 16 bits (8 for piggyback + 8 for exit done) would be set according to whether the exit is piggybacked or done.

I'd like to know what did I miss here?

(BTW, I am thinking to create an issue of adding docs on our contract data structure. How do people think?)

Reply:
@boolafish not answering your question fully, but on documenting the `bitSet` magic we're planning to wrap all such operations in functions with descriptive names, c.f. OMG-409 ticket in pivotal.

Reply:
@boolafish the most significant bit is set when the IFE is finalized completely (the IFE has been processed on each queue it had been pushed to) - I'm just documenting it ;)

Reply:
@boolafish `exitmap` was chosen as a place to have this flag because we can cram into 32 bytes both information about piggybacks status and IFE status. Piggyback status info might be used later (imagine updating the contract to re-enable IFE reactivation).